src/Helpers/CopilotTokenRefreshPolicy.cs

```diff
@@ -0,0 +1,151 @@
+using System;
+using System.ClientModel.Primitives;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+/// <summary>
+/// Policy that automatically refreshes a GitHub Copilot token before it expires.
+/// </summary>
+public class CopilotTokenRefreshPolicy : PipelinePolicy
+{
+    private readonly string _githubToken;
```

Comment:
i've typically been putting class data at the bottom of the C# class... Not above the methods, but below the methods.